pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -256,6 +257,71 @@ func (state PipelineRunState) GetRunsResults() map[string][]v1alpha1.RunResult {
 	return results
 }
 
+// GetChildReferences returns a slice of references, including version, kind, name, and pipeline task name, for all
+// TaskRuns and Runs in the state.
+func (state PipelineRunState) GetChildReferences(taskRunVersion string, runVersion string) []v1beta1.ChildStatusReference {
+	var childRefs []v1beta1.ChildStatusReference
+
+	for _, rprt := range state {
+		if rprt.ResolvedConditionChecks == nil && ((rprt.CustomTask && rprt.Run == nil) || (!rprt.CustomTask && rprt.TaskRun == nil)) {
```

Comment:
can this conditional logic be simplified? especially since [isCustomTask](https://github.com/tektoncd/pipeline/blob/main/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L124-L127) just returns rprt.CustomTask.

One idea is to delete this block, and below have:

```
if rprt.CustomTask {
			if rprt.Run != nil {
				childAPIVersion = rprt.Run.APIVersion
			} else if rprt.ResolvedConditionChecks == nil {
				continue
                         } else {
				childAPIVersion = runVersion
			}
		} else {
			...
}
```

Reply:
Good call!

Reply:
Ok, I think we still need/want that if chunk, but just for `TaskRun`s - `ResolvedConditionChecks` doesn't apply at all for `Run`s, so it's irrelevant there. Tweak incoming.

Reply:
Pushed!